src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsValueSetter.cs

```diff
@@ -586,9 +589,17 @@ internal void SetDateTime(DateTime value)
                     if (SqlDbType.DateTime2 == _metaData.SqlDbType)
                     {
                         long time = value.TimeOfDay.Ticks / TdsEnums.TICKS_FROM_SCALE[_metaData.Scale];
+#if NETCOREAPP
+                        _stateObj.WriteByteArray(BitConverter.GetBytes(BinaryPrimitives.ReadInt64LittleEndian(BitConverter.GetBytes(time))), (int)_metaData.MaxLength - 3, 0);
```

Comment:
I think what you're looking for is something like this:
```
Span<byte> result = stackalloc byte[8];
BinaryPrimitives.WriteInt64LittleEndian(result, time);
_stateObj.WriteByteArray(result.AsSpan(0, (int)_metaData.MaxLength - 3));
```
You can follow this example to fix the rest of the places where you used BitConverter.GetBytes twice. Should address @Wraith2's [concerns](https://github.com/dotnet/SqlClient/pull/2170/commits/4114d8af84b06f519b2f3e1c453f729b39d56d74#r1338730042).

On a side note, this snippet uses 8 bytes from stack instead of 8 bytes from heap, so it seems to be an improvement over the original code in terms of performance.

Reply:
I tried this suggestion it seems to be throwing a compilation error. it seems `Span<byte> doesn't have a overloading method AsSpan()` .  as a turnaround how about this?
```
 unsafe
 {
      var result = stackalloc byte[8];
      BinaryPrimitives.WriteInt64LittleEndian(new Span<byte>(result,8), time);
      _stateObj.WriteByteSpan(new Span<byte>(result,(int)_metaData.MaxLength - 3)); 
}
```
@panoskj

Reply:
I'm also not sure why you added `unsafe`.

Reply:
`result` is a byte * so it should be used in a unsafe context. right?

Reply:
done